firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift

```diff
@@ -0,0 +1,902 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Shared
+import UIKit
+import Common
+import ComponentLibrary
+import SiteImageView
+
+struct TPMenuUX {
+    struct Fonts {
+        static let websiteTitle: UIFont = .systemFont(ofSize: 17, weight: .semibold)
+        static let viewTitleLabels: UIFont = .systemFont(ofSize: 17, weight: .regular)
+        static let detailsLabel: UIFont = .systemFont(ofSize: 12, weight: .regular)
+        static let minorInfoLabel: UIFont = .systemFont(ofSize: 15, weight: .regular)
+    }
+
+    struct UX {
+        static let baseCellHeight: CGFloat = 44
+        static let popoverTopDistance: CGFloat = 20
+        static let horizontalMargin: CGFloat = 16
+        static let viewCornerRadius: CGFloat = 8
+        static let headerLabelDistance: CGFloat = 2.0
+        static let headerLinesLimit: Int = 2
+        static let foxImageSize: CGFloat = 100
+        static let iconSize: CGFloat = 24
+        static let protectionViewBottomSpacing: CGFloat = 70
+        static let siteDomainLabelsVerticalSpacing: CGFloat = 12
+        static let connectionDetailsLabelBottomSpacing: CGFloat = 28
+        static let connectionDetailsHeaderMargins: CGFloat = 8
+        static let faviconImageSize: CGFloat = 40
+        static let closeButtonSize: CGFloat = 30
+        static let faviconCornerRadius: CGFloat = 5
+        static let scrollContentHorizontalPadding: CGFloat = 16
+
+        static let clearDataButtonCornerRadius: CGFloat = 12
+        static let clearDataButtonBorderWidth: CGFloat = 1
+        static let settingsLinkButtonBottomSpacing: CGFloat = 32
+        struct Line {
+            static let height: CGFloat = 1
+        }
+        struct TrackingDetails {
+            static let baseDistance: CGFloat = 20
+            static let imageMargins: CGFloat = 10
+            static let bottomDistance: CGFloat = 350
+        }
+        struct BlockedTrackers {
+            static let headerDistance: CGFloat = 8
+        }
+    }
+}
+
+protocol TrackingProtectionMenuDelegate: AnyObject {
+    func settingsOpenPage(settings: Route.SettingsSection)
+    func didFinish()
```

Comment:
the Redux architecture should replace these delegates, I believe. I also didn't see redux implemented for this VC. Assuming that it will be in a TODO, should we add that somewhere in this view controller?

Reply:
Redux side is done by Stefan and will be merged soon with this controller.